tests/accelerators/test_accelerator_connector.py

```diff
@@ -95,7 +95,8 @@ def test_accelerator_choice_ddp_spawn(cuda_available_mock, device_count_mock):
         "SLURM_NTASKS": "2",
         "SLURM_JOB_NAME": "SOME_NAME",
         "SLURM_NODEID": "0",
-        "SLURM_LOCALID": "10"
+        "SLURM_PROCID": "1",
+        "SLURM_LOCALID": "1",
```

Comment:
Why Do you need to change these?

Reply:
these were artificial values, and they don't match with what we expect. 
SLURM_NTASKS=2 but LOCAL_RANK 10 is not valid, and would actually lead to Lightning ignoring the SlurmEnvironment and instead select the LightningEnvironment (there is a [comment ](https://github.com/PyTorchLightning/pytorch-lightning/blob/b9bc77293bbe324c5b3f7b0d7498a0db011a391a/pytorch_lightning/trainer/connectors/accelerator_connector.py#L606)in the code, we should document that behavior better, and write proper tests instead of these misleading ones that were here)